### PR TITLE
Add openmmforcefields and perses

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Supersedes #13234 

- [x] `openmmforcefields` requires a new `parmed` release -- see https://github.com/conda-forge/parmed-feedstock/pull/9. This blocks `perses`.

### CODE REVIEW

File: recipes/openff-recharge/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{% set name = "openff-recharge" %}
+{% set version = "0.0.1a4" %}
```
Comment (SimonBoothroyd) [inline recipes/openff-recharge/meta.yaml:2]:
```suggestion
{% set version = "0.0.1a6" %}
```

File: recipes/openff-recharge/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/openforcefield/{{ name }}/archive/{{ version | replace("a", "-alpha.") }}.tar.gz
+  sha256: "00a2d2285a6059b3cd4f2c4d64caa261c660f8b071ae27e193353bbb1b5837e2"
```
Comment (SimonBoothroyd) [inline recipes/openff-recharge/meta.yaml:10]:
```suggestion
  sha256: "1800043c61320ae2f90129bbcb66fb897254c62a3301c906867c6016c095664b"
```

File: recipes/openff-recharge/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  entry_points:
+  - recharge = openff.recharge.cli:cli
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python >=3.6
+    - click
```
Comment (SimonBoothroyd) [inline recipes/openff-recharge/meta.yaml:25]:
```suggestion
    - click
    - tqdm
    - cmiles-base
```

File: recipes/openff-recharge/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+test:
+  imports:
+    - openff.recharge
+
+about:
+  home: https://openforcefield.org/
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: An automated framework for generating optimized partial charges for molecules.
```
Comment (SimonBoothroyd) [inline recipes/openff-recharge/meta.yaml:42]:
```suggestion
  summary: An framework for generating partial charges for molecules from QC data.
```

File: recipes/openff-recharge/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  description: |
+    An automated framework for optimising molecular partial charges against electrostatic
+    potential data using bond-charge corrections
```
Comment (SimonBoothroyd) [inline recipes/openff-recharge/meta.yaml:45]:
```suggestion
    A framework which aims to provide utilities for optimising the partial charges of molecules against 
    QC derived electrostatic potential and electric field data.
```

**Review by SimonBoothroyd — CHANGES_REQUESTED** [review]
LGTM after the changes.

File: recipes/openff-recharge/meta.yaml (hunk 1 above)
Comment (jaimergp) [reply recipes/openff-recharge/meta.yaml:2]: Oh, this reminds me I found how to publish alphas / betas / rcs in conda-forge. See https://github.com/conda-forge/cfep/blob/master/cfep-05.md. We might want to "break" recharge on purpose so it doesn't upload anything to `main` upon rendering.

File: recipes/openff-toolkit/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
+{% set name = "openff-toolkit" %}
+{% set version = "0.8.0" %}
```
Comment (mattwthompson) [inline recipes/openff-toolkit/meta.yaml:2]:
```suggestion
{% set version = "0.8.1" %}
```

File: recipes/openff-toolkit/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/openforcefield/openforcefield/archive/{{ version }}.tar.gz
+  sha256: cd4ca0ded60ebd97120523b85c2aa6e0ddbc400193e6bbf49aa52b82ccf9e026
```
Comment (mattwthompson) [inline recipes/openff-toolkit/meta.yaml:10]:
```suggestion
  sha256: 20597c0e325ad76d1bf161eb88717c397af5d1d794f4b694d28784b8e38d7560
```

Assuming I remembered the right way to get the sum:
```
wget https://github.com/openforcefield/openforcefield/archive/0.8.1.tar.gz
sha256sum 0.8.1.tar.gz
```

File: recipes/openff-toolkit/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv --no-deps"
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
```
Comment (mattwthompson) [inline recipes/openff-toolkit/meta.yaml:21]: Also needs `mdtraj` (long story, will be fixed in next release)

**Comment by mattwthompson** [comment]
I agree to be a maintainer of the recipes listed by my name.

**Review by mattwthompson — COMMENTED** [review]
Just a few changes as a result of the 0.8.1 release earlier in the week

Thanks for handling everything!

**Comment by j-wags** [comment]
I agree to be a maintainer of openff-evaluator, openff-forcefields, openff-recharge, openff-smirnoff99frosst, openff-toolkit.

I'd like to keep from publishing these packages on `conda-forge`s `main` label until we return from the holidays in January. I'd be happy to publish on a non-main label for testing sooner.

**Comment by peastman** [comment]
I agree to be a maintainer.

**Comment by jchodera** [comment]
I agree to be a maintainer!

File: recipes/openff-evaluator/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,67 @@
+{% set name = "openff-evaluator" %}
+{% set version = "0.3.1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/openforcefield/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: "4e8f0140b53cbfe6123e6280dfcd5fed1f266fb80cd3b357b8df94c71be20f48"
```
Comment (SimonBoothroyd) [inline recipes/openff-evaluator/meta.yaml:10]:
```suggestion
  sha256: "d36f9dc62ce0571c4fe2caa42a926c9bf8e70759ed661ab292d2b4cc2b8b171b"
```

**Comment by mattwthompson** [comment]
Given that we have some packages being renamed, and users might be installing these into existing environments with differently-named-but-the-same-software packages from `omnia`, should we add some `run_constrained` lines? (Thanks to @SimonBoothroyd for suggesting this!)

**Comment by jaimergp** [comment]
> Should we add some run_constrained lines?

Oh, yes, this should be added!

**Comment by mattwthompson** [comment]
Some serialization deps for the toolkit can be trimmed down now that is https://github.com/openforcefield/openforcefield/pull/794 merged (still pending a release)

**Comment by jaimergp** [comment]
Note this will fail until #13635 is merged

**Comment by jaimergp** [comment]
@jchodera Can you check if the [dependencies](https://github.com/jaimergp/staged-recipes/blob/omnia-migration/recipes/openmmforcefields/meta.yaml#L21-L30)for `openmmforcefields` (and their constraints) are accurate?

**Comment by mikemhenry** [comment]
It looks like the windows build is failing to find `ambertools` but  [ambertools](https://anaconda.org/conda-forge/ambertools) doesn't have any windows builds. If we don't have any intention to support windows for `openmmforcefields` then we can drop the windows builds.

File: recipes/openmmforcefields/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+source:
+  url: https://github.com/openmm/openmmforcefields/archive/{{ version }}.tar.gz
+  sha256: e6e89a5e75b0a1e1609b7d5480793e6862245cfdad9ae46f0f30a66061480355
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv --no-deps"
+
```
Comment (mikemhenry) [inline recipes/openmmforcefields/meta.yaml:16]:
```suggestion
skip: True  # [win]

```

Comment (mikemhenry) [reply recipes/openmmforcefields/meta.yaml:16]: I think this will work to skip windows builds/testing

**Comment by jchodera** [comment]
> @jchodera Can you check if the dependencies for openmmforcefields (and their constraints) are accurate?

@jaimergp: The dependencies appear to be accurate!

If there's no `ambertools` windows build, we can skip win builds for now.

File: recipes/openmmforcefields/meta.yaml (hunk 10 above)
Comment (jaimergp) [reply recipes/openmmforcefields/meta.yaml:16]: We can't use platform selectors for skip if `noarch: python` is set. It is expected that Windows will fail for a noarch if dependencies are missing!

**Comment by jaimergp** [comment]
@conda-forge/help-python-c, this is ready for review, thanks!

File: recipes/perses/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    - mdtraj
+    - pymbar
+    - ambertools >=19.11
+    - openmm >=7.4.2
+    - openmmforcefields
+    - openmoltools >=0.8.4
+    - openmmtools >=0.20.0
+    - yank
+    - pdbfixer
+    # - openeye-toolkits
```
Comment (chrisburr) [inline recipes/perses/meta.yaml:52]: Is there a reason why this isn't being added to conda-forge? (point me to a comment if there already is one)

Comment (jaimergp) [reply recipes/perses/meta.yaml:52]: Licensing issues, sadly They are only available through the [company's channel](https://anaconda.org/openeye/openeye-toolkits). 

We work around that requirement using try-guarded imports for that package, and make sure to mention the extra package in the errors plus documentation.

**Comment by jaimergp** [comment]
Thanks @chrisburr!

**Pull request merged**